### ISSUE TITLE
test set seems to be used in training

### ISSUE THREAD

**Issue by NomotoEriko** [junyanz/pytorch-CycleGAN-and-pix2pix#711]
Hi, 

When I tried to train colorization model on my own dataset, I realized this problem.
My set up is below.
> Directory Tree (Some jpg files are omitted)
```
datasets
└── bird
    ├── test
    │   ├── 1.jpg
    │   ├── 2.jpg
    │   └── 3.jpg
    └── train
        ├── 1.jpg
        ├── 2.jpg
        └── 3.jpg
```

> Command
```
$ python train.py --dataroot ./datasets/bird --name color_pix2pix --model colorization --gpu_ids 1
```

And I found `The number of training images = 11794` in stdout. 11794 is the sum of train set and test set. I did print out `data["A_paths"]` on the start of each iteration (around [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/train.py#L44)), and the test data was found.
> output
```
create web directory ./checkpoints/color_pix2pix/web...
A_paths: ./datasets/bird/train/8704.jpg 
A_paths: ./datasets/bird/test/1822.jpg 
A_paths: ./datasets/bird/train/1974.jpg 
A_paths: ./datasets/bird/train/8509.jpg 
A_paths: ./datasets/bird/train/5519.jpg 
A_paths: ./datasets/bird/train/2951.jpg 
A_paths: ./datasets/bird/train/6761.jpg 
A_paths: ./datasets/bird/train/5797.jpg 
A_paths: ./datasets/bird/train/3674.jpg 
A_paths: ./datasets/bird/train/2131.jpg 
A_paths: ./datasets/bird/train/23.jpg 
A_paths: ./datasets/bird/train/495.jpg 
A_paths: ./datasets/bird/test/2859.jpg 
A_paths: ./datasets/bird/train/5909.jpg 
```
I suppose my setup is not bad according to the [documentation](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/tips.md).

I think there are two solutions for it.
1. Use `./datasets/bird/train` for `--dataset` option instead of `./datasets/bird`
2. Modify [colorization_dataset.py#L39](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/data/colorization_dataset.py#L39) to `self.dir = os.path.join(opt.dataroot, opt.phase)`

According to the documentation and the implementation of AlignedDataset, I suppose the solution-2 is better. If the solution-1 is better, I think the documentation should be updated.

**Comment by junyanz** [comment]
This is a good catch. I just updated the code (solution 2).
